tests/rest/admin/test_room_reports.py

```diff
@@ -0,0 +1,421 @@
+        self.assertEqual(channel.json_body["total"], 4)
+        self.assertEqual(len(channel.json_body["room_reports"]), 2)
+        self.assertNotIn("room_reports", channel.json_body)
+
+        #  `next_token` does not appear
+        # Number of max results is larger than the number of entries
+        channel = self.make_request(
+            "GET",
+            self.url + "?limit=5",
+            access_token=self.admin_user_tok,
+        )
+
+        self.assertEqual(200, channel.code, msg=channel.json_body)
+        self.assertEqual(channel.json_body["total"], 4)
+        self.assertEqual(len(channel.json_body["room_reports"]), 4)
+        self.assertNotIn("next_token", channel.json_body)
+
+        #  `next_token` does appear
+        # Number of max results is smaller than the number of entries
+        channel = self.make_request(
+            "GET",
+            self.url + "?limit=3",
+            access_token=self.admin_user_tok,
+        )
+
+        self.assertEqual(200, channel.code, msg=channel.json_body)
+        self.assertEqual(channel.json_body["total"], 4)
+        self.assertEqual(len(channel.json_body["room_reports"]), 3)
+        self.assertEqual(channel.json_body["next_token"], 3)
+
+        # Check
+        # Set `from` to value of `next_token` for request remaining entries
+        #  `next_token` does not appear
+        channel = self.make_request(
+            "GET",
+            self.url + "?from=3",
+            access_token=self.admin_user_tok,
+        )
+
+        self.assertEqual(200, channel.code, msg=channel.json_body)
+        self.assertEqual(channel.json_body["total"], 4)
+        self.assertEqual(len(channel.json_body["room_reports"]), 1)
+        self.assertNotIn("next_token", channel.json_body)
+
+    def _report_room(self, room_id: str, user_tok: str) -> None:
+        """Report a room"""
+        channel = self.make_request(
+            "POST",
+            "rooms/%s/report" % room_id,
+            {"reason": "this makes me sad"},
+            access_token=user_tok,
+        )
+        self.assertEqual(200, channel.code, msg=channel.json_body)
+
+    def _report_room_without_parameters(self, room_id: str, user_tok: str) -> None:
+        """Report a room, but omit reason"""
+        channel = self.make_request(
+            "POST",
+            "rooms/%s/report" % room_id,
+            {},
+            access_token=user_tok,
+        )
+        self.assertEqual(200, channel.code, msg=channel.json_body)
+
+    def _check_fields(self, content: List[JsonDict]) -> None:
+        """Checks that all attributes are present in a room report"""
+        for c in content:
+            self.assertIn("id", c)
+            self.assertIn("received_ts", c)
+            self.assertIn("room_id", c)
+            self.assertIn("user_id", c)
+            self.assertIn("canonical_alias", c)
+            self.assertIn("name", c)
+            self.assertIn("reason", c)
```

Comment:
To make sure we don't accidentally introduce another field (since we're taking raw output from a database storage function:

```suggestion
            self.assertIn("reason", c)
        
        self.assertEqual(len(c.keys()), 7)
```

Reply:
(edited for syntax)